sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -23,6 +23,40 @@ public partial class AzureKeyCredential
         public void Update(string key) { }
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct BinaryData
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        public BinaryData(byte[] data) { throw null; }
+        public BinaryData(object serializable, Azure.Core.Serialization.ObjectSerializer serializer, System.Type? type = null) { throw null; }
+        public BinaryData(object jsonSerializable, System.Type? type = null) { throw null; }
+        public BinaryData(System.ReadOnlyMemory<byte> data) { throw null; }
+        public BinaryData(System.ReadOnlySpan<byte> data) { throw null; }
+        public BinaryData(string data) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object? obj) { throw null; }
```

Comment:
Should we even override this one? It's a allocation to even call it. If we think feature is important we should implement a strongly typed version of it.

Reply:
And maybe we need to override gethashcode as well.

Reply:
We do override GetHashCode.

One benefit of overriding this one is that we can at least document what the behavior is for users so that they understand that it is not using value equality.